packages/storage/src/cookies.ts

```diff
@@ -58,20 +61,28 @@ try {
  *   httpOnly?: boolean;
  *   maxAge?: number;
  *   sameSite?: "None" | "Lax" | "Strict";
+ *   getRequest?: () => Request | () => PageEvent // (useRequest from solid-start)
+ *   setCookie?: (key, value, options) => void // set cookie on the server
  * };
  * ```
  * Also, you can use its _read and _write properties to change reading and writing
  */
 export const cookieStorage: StorageWithOptions<CookieOptions> = addClearMethod({
-  _read: isServer ? () => useRequest().request.headers.get("Cookie") : () => document.cookie,
+  _read: isServer
+    ? (options?: CookieOptions) => {
+      const eventOrRequest = options?.getRequest?.() || useRequest?.();
+      const request = eventOrRequest && ('request' in eventOrRequest ? eventOrRequest.request : eventOrRequest);
+      return request?.headers.get("Cookie") || "";
```

Comment:
For more flexibility, the new option should return the whole Cookie string, because other framework's request object might not use `request.headers.get("Cookie")`

Reply:
Which solid-based framework gives you the cookies as string and doesn't provide an easy access to the Request object?

Reply:
> Which solid-based framework gives you the cookies as string and doesn't provide an easy access to the Request object? 

I don't have a concrete example, it's only a possibility.

Reply:
Thank you for considering that. I'm hesitant to add stuff that's very probably never going to be used.